runComputeDFAPath_algorithm.py

```diff
@@ -0,0 +1,243 @@
+
+        self.addOutput(QgsProcessingOutputVectorLayer(
+            self.OUTPUT,
+            self.tr("Output layer"),
+            QgsProcessing.TypeVectorAnyGeometry))
+
+        self.addOutput(QgsProcessingOutputMultipleLayers(
+                self.OUTPPR,
+            ))
+
+    def flags(self):
+        return super().flags()
+        # return super().flags() | QgsProcessingAlgorithm.FlagNoThreading
+
+    def processAlgorithm(self, parameters, context, feedback):
+        """
+        Here is where the processing itself takes place.
+        """
+
+        import avaframe.version as gv
+        from . import avaframeConnector_commonFunc as cF
+
+        feedback.pushInfo('AvaFrame Version: ' + gv.getVersion())
+
+        # targetADDTONAME = self.parameterAsString(parameters, self.ADDTONAME, context)
+        targetADDTONAME = ''
+
+        sourceDEM = self.parameterAsRasterLayer(parameters, self.DEM, context)
+        if sourceDEM is None:
+            raise QgsProcessingException(self.invalidSourceError(parameters, self.DEM))
+
+        # Release files
+        allREL = self.parameterAsLayerList(parameters, self.REL, context)
+        if allREL is None:
+            raise QgsProcessingException(self.invalidSourceError(parameters, self.REL))
+
+        relDict = {}
+        if allREL:
+            relDict = {lyr.source(): lyr for lyr in allREL}
+
+        #runDFA bool
+        sourceRUNDFA = self.parameterAsBool(parameters, self.RUNDFA, context)
+
+        sourceFOLDEST = self.parameterAsFile(parameters, self.FOLDEST, context)
+
+        # create folder structure (targetDir is the tmp one)
+        finalTargetDir, targetDir = cF.createFolderStructure(sourceFOLDEST)
+
+        feedback.pushInfo(sourceDEM.source())
+
+        # copy DEM
+        cF.copyDEM(sourceDEM, targetDir)
+
+        # copy all release shapefile parts
+        cF.copyMultipleShp(relDict, targetDir / 'Inputs' / 'REL', targetADDTONAME)
+
+        feedback.pushInfo('Starting path generation')
+        feedback.pushInfo('See console for progress')
+
+        # Prepare command for subprocess call
+        command = ['python', '-m', 'avaframe.runComputeDFAPath', str(targetDir)]
+
+        # Add --runDFA argument if RUNDFA boolean is True
+        if sourceRUNDFA:
+            command.append('--runDFA')
+
+        subprocess.call(command)
+        feedback.pushInfo('Done, start loading the results')
+
+        # Move input, log and output folders to finalTargetDir
+        cF.moveInputAndOutputFoldersToFinal(targetDir, finalTargetDir)
+
+        # Get path and split point shapefiles to return to QGIS
+        try:
+            pathResults = cF.getDFAPathResults(finalTargetDir)
+        except:
+            raise QgsProcessingException(self.tr('Something went wrong with path generation, please check log files'))
+
+        context = cF.addLayersToContext(context, pathResults, self.OUTPUT)
+
+        # If runDFA, try to get eventual peakfiles from com1DFA to return to QGIS
```

Comment:
remove this comment block. This not a feature we need for now

Reply:
done